From 88707f2131597022bfdd20fad258de4f45b10dd9 Mon Sep 17 00:00:00 2001
From: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Date: Fri, 24 Mar 2017 19:13:27 +0900
Subject: [PATCH 549/909] arm64: dts: r8a7795: Add cpu capacity-dmips-mhz

commit 1f213836d1de2123a7fe4deb9eb5829a72c81fda from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Set the capacity-dmips-mhz for R-CAR H3, that is based on dhrystone.

Expected cpu capacity:
Cortex-A57: 1024, Cortex-A53: 432

Note:
The cpu capacity in Cortex-A53 side is scaled with max cpufreq
in current implementation.

540 * 1.2[GHz]/1.5[GHz] = 432

Therefore, it needs to set "540" to capacity-dmips-mhz in dt
if it uses cpu capacity with "432" on Cortex-A53.

Signed-off-by: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index a3b9a47c5813..4a481d19094e 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -406,6 +406,7 @@
 				<&cluster0_opp5>, <&cluster0_opp6>,
 				<&cluster0_opp7>;
 			#cooling-cells = <2>;
+			capacity-dmips-mhz = <1024>;
 		};
 
 		a57_1: cpu@1 {
@@ -423,6 +424,7 @@
 				<&cluster0_opp5>, <&cluster0_opp6>,
 				<&cluster0_opp7>;
 			#cooling-cells = <2>;
+			capacity-dmips-mhz = <1024>;
 		};
 
 		a57_2: cpu@2 {
@@ -440,6 +442,7 @@
 				<&cluster0_opp5>, <&cluster0_opp6>,
 				<&cluster0_opp7>;
 			#cooling-cells = <2>;
+			capacity-dmips-mhz = <1024>;
 		};
 
 		a57_3: cpu@3 {
@@ -457,6 +460,7 @@
 				<&cluster0_opp5>, <&cluster0_opp6>,
 				<&cluster0_opp7>;
 			#cooling-cells = <2>;
+			capacity-dmips-mhz = <1024>;
 		};
 
 		a53_0: cpu@100 {
@@ -469,6 +473,7 @@
 			cpu-idle-states = <&CPU_SLEEP_1>;
 			clocks = <&cpg CPG_CORE R8A7795_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			capacity-dmips-mhz = <540>;
 		};
 
 		a53_1: cpu@101 {
@@ -481,6 +486,7 @@
 			cpu-idle-states = <&CPU_SLEEP_1>;
 			clocks = <&cpg CPG_CORE R8A7795_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			capacity-dmips-mhz = <540>;
 		};
 
 		a53_2: cpu@102 {
@@ -493,6 +499,7 @@
 			cpu-idle-states = <&CPU_SLEEP_1>;
 			clocks = <&cpg CPG_CORE R8A7795_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			capacity-dmips-mhz = <540>;
 		};
 
 		a53_3: cpu@103 {
@@ -505,6 +512,7 @@
 			cpu-idle-states = <&CPU_SLEEP_1>;
 			clocks = <&cpg CPG_CORE R8A7795_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			capacity-dmips-mhz = <540>;
 		};
 
 		L2_CA57: cache-controller-0 {
-- 
2.17.1

